core/src/jsMain/kotlin/Scanner.kt

```diff
@@ -31,9 +32,7 @@ public class JsScanner internal constructor(
 
         val scan = bluetooth.requestLEScan(options.toDynamic()).await()
         val listener: (Event) -> Unit = {
-            runCatching {
-                offer(Advertisement(it as BluetoothAdvertisingEvent))
-            }.onFailure {
+            trySend(Advertisement(it as BluetoothAdvertisingEvent)).getOrElse {
                 console.warn("Unable to deliver advertisement event due to failure in flow or premature closing.")
```

Comment:
Ditto, do we want to print the exception too?

Reply:
What is the standard way to log an exception in JavaScript?

Reply:
For production it's best not to log at all and rely on `throw`, but this is probably fine.